### ISSUE TITLE
ENH: Implement a more dynamic `delay`.

### ISSUE THREAD

**Issue by tobiasraabe** [pytask-dev/pytask-parallel#42]
- Add it to the readme.
- Maybe decrease the default value since going (not executing) 1200 tasks with 4 workers will take at least 1200 / 4 * 0.1 = 30s just due to the delay.
- Maybe increase sleep time slowly until 1s when no new tasks is scheduled and one is waiting on the running tasks. Drop to the default if another task was scheduled.
